amethyst_assets/src/merge.rs

```diff
@@ -1,68 +0,0 @@
-// FIXME: This is almost a copy for `Merge` from _specs_. Should be moved there
-
-use std::error::Error;
-use std::marker::PhantomData;
-
-use futures::{Async, Future};
-use specs::{Component, Entity, Entities, Join, RunningTime, System, WriteStorage};
-
-
-/// A system which merges `Ready` futures into the persistent storage.
-/// Please note that your `World` has to contain a component storage
-/// for `F` and `F::Item`.
-///
-/// In case of an error, it will be added to the `Errors` resource.
-pub struct Merge<F> {
-    future_type: PhantomData<*const F>,
-    tmp: Vec<Entity>,
-}
-
-unsafe impl<F> Send for Merge<F> {}
-unsafe impl<F> Sync for Merge<F> {}
-
-impl<F> Merge<F> {
-    /// Creates a new merge system.
-    pub fn new() -> Self {
-        Merge {
-            future_type: PhantomData,
-            tmp: Vec::new(),
-        }
-    }
-}
-
-impl<'a, T, E, F> System<'a> for Merge<F>
-    where T: Component + Send + Sync + 'static,
-          F: Future<Item = T, Error = E> + Component + Send + Sync,
-          E: Error,
-{
-    type SystemData = (Entities<'a>,
-                       // FIXME: Here should be some Fetch<'a, Log> to log occured errors
```

Comment:
That's exactly why we cannot have generic errors; there has to be one single place in the ECS for all errors, and that's [`specs::common::Errors`](https://github.com/slide-rs/specs/blob/master/src/common.rs#L71). Please note though that raising errors in the ECS should, in general, be avoided.